storybookjs/storybook
# CLI: Reuse existing chromium tab if possible

Issue: #10328 

## What I did
This PR moves from the library open to better-opn. What it does is basically tries to use the well known solution [started at CRA](https://github.com/facebook/create-react-app/blob/master/packages/react-dev-utils/openBrowser.js), with an applescript to check if there is an existing tab on chrome and if so, reuses it. As a fallback, it uses the `open` lib.

This feature is unfortunately available on OSX Chrome only. `better-opn` is entirely based in the solution from facebook, and actually they recently added support for chromium based browsers [here](https://github.com/facebook/create-react-app/commit/d2de54b25cc25800df1764058997e3e274bd79ac). I might add a PR on `better-opn` to update that as well.

The alternatives to support reusing tabs are:

**instead of using:**
`import open from 'better-opn'`
**use:**
`import open from 'react-dev-utils/openBrowser'`
However they do not recommend to use their internal tools directly because they can always change it without notice. In this particular case, I guess the chances are small though.

## How to test

On OSX, checkout the branch and run `yarn start`, then kill the process and run it again.
You should see the tab being reused.
I couldn't test on windows, but as the lib falls back to `open`, it should behave the same as before, but of course if someone could test it out would be great.

closes #10328
